### DISCUSSION TITLE
Integrated terminal not working on iPad OS

### DISCUSSION THREAD

**Discussion by josh-unwin** [community/community#4760] [Codespaces] [answered]
Hi there,

I’m trying out Codespaces as an option for emergency work on my iPad when I’m away from my laptop, but I’m finding text isn’t rendering in the integrated terminal on both Safari or Chrome on iPad OS (I’m aware all browsers use Safaris WebKit under the hood).

I’ve tried it on two different repos/codespaces.

Using iPad OS 15 beta 4.

See attached screenshot below.
<img width="1194" alt="PNG image 5" src="https://user-images.githubusercontent.com/49525488/126690676-9f3d61ed-16b9-4510-a566-369e1ed6300a.png">

**Comment by tanmayeekamath** [comment]
Hey @josh-unwin thanks for reporting this. We are investigating the issue here, I'll report back with an update soon.

**Comment by huw** [comment]
I'm seeing this issue too—Safari 15 Technology Preview, macOS Big Sur 11.4 (but not Safari 14). The text can be copied, and it's clearly showing up in the DOM. Just not visible.

**Comment by tanmayeekamath** [comment, marked as answer]
Hey @josh-unwin, @huw 

Can you try by updating the following setting to` "terminal.integrated.gpuAcceleration": "off"` ?
Here is a link to why that might help: https://code.visualstudio.com/docs/editor/integrated-terminal#_why-is-my-terminal-showing-a-multicolored-triangle-or-a-completely-black-rectangle

If this does not fix the issue, please create an issue in the VS Code repo and the team can help investigate: https://github.com/microsoft/vscode/issues/new/choose

Hope this helps :)

**Reply by huw** [reply]
This worked! Thanks so much. Happened to solve a totally unrelated memory leak I was also facing with font rendering—I suspect Apple messed something up with font rendering in the new OS.

**Reply by superzadeh** [reply]
Confirming as well, had the same issue and this fixed it, thanks a lot!

**Reply by michaelvollrath** [reply]
This solved it for me on iPad as well. Thank you so much!

**Reply by anandreddyks** [reply]
Solved my issue using Github codespace on iPad. Thank you !

**Comment by stratgt10** [comment]
Nice

**Comment by jozefvaclavik** [comment]
On ipados 15 you can also switch to `VS Code Insiders`. There terminal works just fine.

**Comment by OccultSlolem** [comment]
> Hey @josh-unwin, @huw 
> 
> 
> 
> Can you try by updating the following setting to` "terminal.integrated.gpuAcceleration": "off"` ?
> 
> Here is a link to why that might help: https://code.visualstudio.com/docs/editor/integrated-terminal#_why-is-my-terminal-showing-a-multicolored-triangle-or-a-completely-black-rectangle
> 
> 
> 
> If this does not fix the issue, please create an issue in the VS Code repo and the team can help investigate: https://github.com/microsoft/vscode/issues/new/choose
> 
> 
> 
> Hope this helps :)

One year on and this still fixes the issue. Thanks!
